sonar-auth-saml-plugin/src/main/java/org/sonarsource/auth/saml/Cookies.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SAML 2.0 Authentication for SonarQube
+ * Copyright (C) 2018-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.auth.saml;
+
+import java.util.Arrays;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is a copy of org.sonar.server.authentication.Cookies from SonarQube codebase
+ *
+ * TODO When using SonarQube 7.3 (https://jira.sonarsource.com/browse/SONAR-11072), this class should be removed
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.auth.saml%3Asonar-auth-saml&pullRequest=4&issues=AWTV8Tvlrfy0B4Fa_mOL&open=AWTV8Tvlrfy0B4Fa_mOL)

[ISSUE_KEY]: AWTV8Tvlrfy0B4Fa_mOL